### PR TITLE
Add backupninja snmp script

### PR DESCRIPTION
**Pull request by AnaelMobilia**
See https://github.com/librenms/librenms/pull/11010

Add ability to monitor last backupninja backup return code.

Thanks to [PVNum](http://www.pvnum.com) for providing time to do.

### CODE REVIEW

File: snmp/backupninja.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python
+import io
+import re
+import gzip
+import os
+import json
+
+LOGFILES = [
+    "/var/log/backupninja.log",
+    "/var/log/backupninja.log.1.gz",
```
Comment (SourceDoctor) [inline snmp/backupninja.py:10]: why opening archived logfile?
poller runs every 5 minutes ...

File: snmp/backupninja.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+def readlog(logfile):
+    if logfile.endswith('.gz'):
+        return gzip.open(logfile,'r')
+    else:
+        return io.open(logfile,'r')
+
+
+def get_backupninja_state():
+    global last_actions, last_fatal, last_error, last_warning
```
Comment (SourceDoctor) [inline snmp/backupninja.py:33]: using global is not such elegant,

better:

```
def get_backupninja_state():
    last_actions = 0
    last_fatal = 0
    last_error = 0
    last_warning = 0
 ... do some stuff ...
    return last_actions, last_fatal, last_error, last_warning
```

and in main:
`last_actions, last_fatal, last_error, last_warning = get_backupninja_state()`

File: snmp/backupninja.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+# Last status
+last_actions = 0
+last_fatal = 0
+last_error = 0
+last_warning = 0
+
+def main():
+    get_backupninja_state()
+    output = {'actions': last_actions, 'fatal': last_fatal, 'error': last_error, 'warning': last_warning}
```
Comment (SourceDoctor) [inline snmp/backupninja.py:21]: your json isn't complete/valid for LibreNMS

structure:
version: (1 in your case, so later upgrades are possible)
error: (by default 0, but can have values on error)
error_string: (by default '', but can contain error message on error)
data: <your dictionary with data>

File: snmp/backupninja.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python
```
Comment (SourceDoctor) [inline snmp/backupninja.py:1]: please use Python3

**Comment by AnaelMobilia** [comment]
@SourceDoctor I updated the code, thanks for the review !

File: snmp/backupninja.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import os
+import json
+
+LOGFILES = [
+    "/var/log/backupninja.log",
+]
+
+def main():
+    last_actions, last_fatal, last_error, last_warning = get_backupninja_state()
+    output = {'version': '1', 'error': '0', 'error_string': '', 'actions': last_actions, 'fatal': last_fatal, 'error': last_error, 'warning': last_warning}
```
Comment (SourceDoctor) [inline snmp/backupninja.py:14]: ok you missunderstood me:
json to be transfered has in general structure:
version   (versionnumber of running agent, starts with 1)
error  (error code from agent, 0 -> no error)
error_string (Errormessage, '' if no error)
data (list or dictionary with your data in here)

e.G.
```
{
version: 1,
error:0
errorstring: '',
data: {'a': 1, 'b: 2 ; 'sizes': ['kevin: 'big', 'stuart': 'medium', 'bob': 'small'], ...}
}
```

in data you're free to build your structure as you need it to transfer it to poller :)

Comment (AnaelMobilia) [reply snmp/backupninja.py]: Thanks for the explanation !
I updated the code :-)

**Pull request merged**